### PR TITLE
Adding oom_score_adj as a container config param

### PR DESCRIPTION
**Pull request by vishh**

### CODE REVIEW

**Comment by vishh** [comment]
cc @mrunalp

File: libcontainer/configs/config.go [hunk 1]
```diff
@@ -127,6 +127,12 @@ type Config struct {
 	// If Rlimits are not set, the container will inherit rlimits from the parent process
 	Rlimits []Rlimit `json:"rlimits"`
 
+	// OomScoreAdj specifies the adjustment to be made by the kernel when calculating oom scores
+	// for a process. Valid values are between the range [-1000, '1000'], where processes with
+	// higher scores are preferred for being killed.
+	// More information about kernel oom score calculation here: https://lwn.net/Articles/317814/
```
Comment (mrunalp) [inline libcontainer/configs/config.go]: I think could point to https://www.kernel.org/doc/Documentation/filesystems/proc.txt instead as oom_score_adj can't be found on the lwn page.

**Comment by mrunalp** [comment]
Also, you might want to propose this to the spec to expose it in the config :)

File: libcontainer/configs/config.go (hunk 1 above)
Comment (vishh) [inline libcontainer/configs/config.go]: Good suggestion. Updated!

**Comment by vishh** [comment]
Yes. The next PR will be add this to the Spec.

File: libcontainer/integration/exec_test.go [hunk 2]
```diff
@@ -878,3 +878,45 @@ func TestMountCgroupRW(t *testing.T) {
...
+		Stdout: &stdout,
+	}
+	err = container.Start(&pconfig)
+	ok(t, err)
+
+	// Wait for process
+	waitProcess(&pconfig, t)
+	outputOomScoreAdj := strings.TrimSpace(string(stdout.Bytes()))
+
+	// Check that the groups output has the groups that we specified
```
Comment (mrunalp) [inline libcontainer/integration/exec_test.go]: Comment needs to be updated.

**Comment by wking** [comment]
On Fri, Aug 28, 2015 at 05:32:38PM -0700, Vish Kannan wrote:
> Yes. The next PR will be add this to the Spec. 

Is a spec entry how we want to handle this?  See
opencontainers/specs#115, where I suggest using a pre-start hook if
possible.  I just haven't gotten around to testing if #160 will make
it possible or not (although see also my uncertainty about the hook
environment [1]).

[1]: https://github.com/opencontainers/runc/pull/160#issuecomment-133091157

File: libcontainer/integration/exec_test.go (hunk 2 above)
Comment (vishh) [inline libcontainer/integration/exec_test.go]: Done. Good catch :)

**Comment by vishh** [comment]
Ping @crosbymichael @LK4D4

**Comment by LK4D4** [comment]
LGMT

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**